src/Microsoft.DotNet.Configurer/NuGetCacheSentinel.cs

```diff
@@ -23,7 +23,7 @@ private string NuGetCachePath
             {
                 if (string.IsNullOrEmpty(_nugetCachePath))
                 {
-                    _nugetCachePath = PackageDependencyProvider.ResolvePackagesPath(null, null);
+                    _nugetCachePath = NuGetPathContext.Create(new NullSettings()).UserPackageFolder;
```

Comment:
it is called at the beginning of every CLI command:

https://github.com/dotnet/cli/blob/rel/1.0.0-preview2/src/dotnet/Program.cs#L169

And has no idea yet of a "project.json".  In some cases you won't even have a project.json, becuase this is called on `dotnet --version`, `dotnet --info` and `dotnet --help`.  So as @brthor points out, CurrentDirectory would be the only thing you could get at this point, which is incorrect IMO. For example, when VS or VS Code run CLI for the first time on the machine, what CurrentDirectory does it use?

/cc @livarcocc - who implemented this class.